src/service/data_products/common_functions.py

```diff
@@ -109,6 +112,78 @@ async def get_collection_singleton_from_db(
     )
 
 
+async def get_columnar_attribs_meta(
+        storage: ArangoStorage,
+        collection: str,
+        collection_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        return_only_visible: bool = False
+) -> col_models.ColumnarAttributesMeta:
+    """
+    Get the columnar attributes meta document for a collection. The document is expected to be
+    a singleton per collection load version.
+
+    storage - the storage system.
+    collection - the arango collection containing the document.
+    collection_id - the KBase collection containing the document.
+    load_ver - the load version of the collection.
+    load_ver_override - whether to override the load version.
+    return_only_visible - whether to return only visible columns.
+
+    """
+    doc = await get_collection_singleton_from_db(
+            storage,
+            collection,
+            collection_id,
+            load_ver,
+            bool(load_ver_override)
+    )
+
+    doc[col_models.FIELD_COLUMNS] = [
+        col_models.AttributesColumn(**d) for d in doc[col_models.FIELD_COLUMNS] if
+        not return_only_visible or not d[col_models.NON_VISIBLE]
```

Comment:
```suggestion
        col_models.AttributesColumn(**d) for d in doc[col_models.FIELD_COLUMNS]
        if not return_only_visible or not d[col_models.NON_VISIBLE]
```

Trivial but slightly easier to read IMO